waiter/src/waiter/client_tools.clj

```diff
@@ -570,11 +568,29 @@
          test-prefix (System/getenv "WAITER_TEST_PREFIX")]
      (str/replace (str test-prefix service-name username (rand-int 3000000)) #"-" ""))))
 
-(defn delete-token-and-assert
-  [waiter-url token & {:keys [hard-delete] :or {hard-delete true}}]
-  (log/info "deleting token" token)
+(defn token->etag
```

Comment:
Typical client behavior:
 1.  Get token data
 2.  Inspect token data, making changes after having observed the token data
 3.  Post the modifications along with the ETag received in Step 1.

Having a function like `attach-token-etag` means we don't act like a client.  `attach-token-etag` feels like it's being used to simply blow away/force-write the token data intentionally avoiding the concurrency mechanism, and hence, I'm not a fan.

Reply:
Done.